`--no-server` doesn't seem to work

## steps

sbt 1.11.3, installed with sbt/setup-sbt

https://github.com/durban/choam/actions/runs/16498673583/job/46650873881#step:17:33

## problem

```
  sbt -v --no-server githubWorkflowCheck
  shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
  env:
    GITHUB_TOKEN: ***
    JAVA_HOME: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\21.0.8-9.0.LTS\arm64
    JAVA_HOME_21_ARM64: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\21.0.8-9.0.LTS\arm64
[sbt_options] declare -a sbt_options=()
[process_args] java_version = '21'
[copyRt] java9_rt = 'C:\Users\runneradmin\.sbt\1.0\java9-rt-ext-eclipse_adoptium_21_0_8/rt.jar'
copying runtime jar...
# Executing command line:
java
-Xms512M
-Xmx6G
-Xss2M
-XX:+UseG1GC
-Ddev.tauri.choam.stats=true
-Djdk.graal.Dump=
-Djdk.graal.PrintBackendCFG=true
-Djdk.graal.MethodFilter=AbstractHamt.*
-Dsbt.io.virtual=false
-Dsbt.server.autostart=false
-Dsbt.script=/c/hostedtoolcache/windows/sbt/1.10.11/sbt/bin/sbt
-Dscala.ext.dirs=C:\Users\runneradmin\.sbt\1.0\java9-rt-ext-eclipse_adoptium_21_0_8
-Djline.terminal=jline.UnixTerminal
-Dsbt.cygwin=true
-jar
C:\hostedtoolcache\windows\sbt\1.10.11\sbt\bin\sbt-launch.jar
githubWorkflowCheck

sbt.internal.ServerAlreadyBootingException: java.io.IOException: Could not create lock for \\.\pipe\sbt-load-1056922235554930886_lock, error 998
sbt thinks that server is already booting because of this exception:
	at sbt.internal.BootServerSocket.newSocket(BootServerSocket.java:357)
	at sbt.internal.BootServerSocket.<init>(BootServerSocket.java:296)
	at sbt.xMain$.getSocketOrExit(Main.scala:153)
	at sbt.xMain$.bootServerSocket$lzycompute$1(Main.scala:79)
	at sbt.xMain$.bootServerSocket$1(Main.scala:79)
	at sbt.xMain$.withStreams$1(Main.scala:87)
	at sbt.xMain$.run(Main.scala:124)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at sbt.internal.XMainConfiguration.run(XMainConfiguration.java:59)
	at sbt.xMain.run(Main.scala:48)
	at xsbt.boot.Launch$.$anonfun$run$1(Launch.scala:149)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:176)
	at xsbt.boot.Launch$.run(Launch.scala:149)
	at xsbt.boot.Launch$.$anonfun$apply$1(Launch.scala:44)
	at xsbt.boot.Launch$.launch(Launch.scala:159)
	at xsbt.boot.Launch$.apply(Launch.scala:44)
	at xsbt.boot.Launch$.apply(Launch.scala:21)
	at xsbt.boot.Boot$.runImpl(Boot.scala:78)
	at xsbt.boot.Boot$.run(Boot.scala:73)
	at xsbt.boot.Boot$.main(Boot.scala:21)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.io.IOException: Could not create lock for \\.\pipe\sbt-load-1056922235554930886_lock, error 998
	at org.scalasbt.ipcsocket.Win32NamedPipeServerSocket.<init>(Win32NamedPipeServerSocket.java:129)
	at org.scalasbt.ipcsocket.Win32NamedPipeServerSocket.<init>(Win32NamedPipeServerSocket.java:48)
	at sbt.internal.BootServerSocket.newSocket(BootServerSocket.java:351)
	... 21 more

Error: Process completed with exit code 2.
```

## expectation

I'd expect that with `--no-server` it doesn't try to create a server(?).

## notes

Related: #6530, #6728, #6101.

Non-deterministic.

--- a/main/src/main/scala/sbt/Main.scala
+++ b/main/src/main/scala/sbt/Main.scala
@@ -77,14 +77,9 @@ private[sbt] object xMain:
       val isBsp: String => Boolean = cmd => (cmd == "-bsp") || (cmd == "--bsp")
       val isNew: String => Boolean = cmd => (cmd == "new") || (cmd == "init")
       lazy val isServer = !userCommands.exists(c => isBsp(c) || isClient(c))
-      // Check if server autostart is disabled via sbt.server.autostart property
-      // This prevents creating boot server socket when --no-server is used
-      lazy val serverAutostart = sys.props.get("sbt.server.autostart") match {
-        case Some(value) => value.toLowerCase == "true"
-        case None        => true // Default to true if not set
-      }
       // keep this lazy to prevent project directory created prematurely
-      lazy val bootServerSocket = if (isServer && serverAutostart) getSocketOrExit(configuration) match {
+      // Only create boot server socket if server mode is enabled and server autostart is enabled
+      lazy val bootServerSocket = if (isServer && SysProp.serverAutoStart) getSocketOrExit(configuration) match {
         case (_, Some(e)) => boundary.break(e)
         case (s, _)       => s
       }

--- a/main/src/main/scala/sbt/Main.scala
+++ b/main/src/main/scala/sbt/Main.scala
@@ -79,11 +79,12 @@ private[sbt] object xMain:
       lazy val isServer = !userCommands.exists(c => isBsp(c) || isClient(c))
       // keep this lazy to prevent project directory created prematurely
       // Only create boot server socket if server mode is enabled and server autostart is enabled
-      lazy val bootServerSocket = if (isServer && SysProp.serverAutoStart) getSocketOrExit(configuration) match {
-        case (_, Some(e)) => boundary.break(e)
-        case (s, _)       => s
-      }
-      else None
+      lazy val bootServerSocket =
+        if (isServer && SysProp.serverAutoStart) getSocketOrExit(configuration) match {
+          case (_, Some(e)) => boundary.break(e)
+          case (s, _)       => s
+        }
+        else None
       lazy val detachStdio = userCommands.exists(_ == BasicCommandStrings.DashDashDetachStdio)
       def withStreams[A](f: => A): A =
         try {

--- a/main/src/main/scala/sbt/Main.scala
+++ b/main/src/main/scala/sbt/Main.scala
@@ -77,8 +77,14 @@ private[sbt] object xMain:
       val isBsp: String => Boolean = cmd => (cmd == "-bsp") || (cmd == "--bsp")
       val isNew: String => Boolean = cmd => (cmd == "new") || (cmd == "init")
       lazy val isServer = !userCommands.exists(c => isBsp(c) || isClient(c))
+      // Check if server autostart is disabled via sbt.server.autostart property
+      // This prevents creating boot server socket when --no-server is used
+      lazy val serverAutostart = sys.props.get("sbt.server.autostart") match {
+        case Some(value) => value.toLowerCase == "true"
+        case None        => true // Default to true if not set
+      }
       // keep this lazy to prevent project directory created prematurely
-      lazy val bootServerSocket = if (isServer) getSocketOrExit(configuration) match {
+      lazy val bootServerSocket = if (isServer && serverAutostart) getSocketOrExit(configuration) match {
         case (_, Some(e)) => boundary.break(e)
         case (s, _)       => s
       }

--- a/main/src/main/scala/sbt/internal/SysProp.scala
+++ b/main/src/main/scala/sbt/internal/SysProp.scala
@@ -100,6 +100,7 @@ object SysProp {
   def legacyTestReport: Boolean = getOrFalse("sbt.testing.legacyreport")
   def semanticdb: Boolean = getOrFalse("sbt.semanticdb")
   def forceServerStart: Boolean = getOrFalse("sbt.server.forcestart")
+  def serverAutoStart: Boolean = getOrTrue("sbt.server.autostart")
   def remoteCache: Option[URI] = sys.props
     .get("sbt.remote_cache")
     .map(URI(_))

